<launch>
  <!-- Configure general settings -->
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find training_coordinator)/config/ros/logging.conf"/>
  <env name="ROSCONSOLE_FORMAT" value="[ ${severity} ${time:%Y-%m-%d %H:%M:%S} ]: ${message}" />
  <param name="use_sim_time" value="True"/>

  <!-- Provide Flatland simulation environment (2D, without agents) -->
  <include file="$(find training_coordinator)/launch/environments/flatland.launch">
    <arg name="show_visualization" value="true"/>
    <arg name="world" default="map_empty"/>
  </include>

  <!-- Training coordinator -->
  <node name="training_coordinator" pkg="training_coordinator" type="train.py" output="screen">
    <param name="pedestrian_model_path" value="$(find training_coordinator)/config/environment/models/pedestrian.yaml" />
    <param name="max_steps_per_episode" value="200" />
    <param name="behavior_model_output_path" value="$(find training_coordinator)/behavior_models/model" />
    <param name="learning_rate" value="3e-3" />
    <param name="total_timesteps" value="10000" />
  </node>

</launch>
